yamx/representer/rendering.py

```diff
@@ -124,7 +124,10 @@ def render_conditional_map(
 
     # TODO: move sorting to different place
     if sort_keys:
-        iter_data = enumerate(sorted(data.items(), key=lambda kv: kv[0]))
+        iter_data = enumerate(
+            # make sure to use original key for sorting
+            sorted(data.items(), key=lambda kv: kv[0].split(DEDUPLICATOR)[0])
```

Comment:
Why is data_key used as a sorting key and not int(UNIQUE_CNT)?

Reply:
`data_key` represents final key which you will see after deduplication, and we want to sort fields using it
`int(UNIQUE_CNT)` is used only to be able to dump identical fields in the same mapping, otherwise dicts like
```
{"field":1, "field":2}
```
aren't valid